### PR TITLE
Fixes podman-remote save to directories does not work

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Fixes: https://github.com/containers/podman/issues/9742
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

File: pkg/api/handlers/libpod/images.go [hunk 1]
```diff
@@ -244,6 +244,7 @@ func ExportImages(w http.ResponseWriter, r *http.Request) {
 	decoder := r.Context().Value("decoder").(*schema.Decoder)
 	query := struct {
 		Compress   bool     `schema:"compress"`
+		Remote     bool     `schema:"remote"`
```
Comment (rhatdan) [inline pkg/api/handlers/libpod/images.go:247]: Why is this needed.  It is Remote by definition?

File: pkg/api/handlers/libpod/images.go [hunk 2]
```diff
@@ -270,6 +271,18 @@ func ExportImages(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// if format is dir and use podman-remote to request, should save to an archive
+	// client will unArchive after receive the archive file
+	// so must convert is at here
+	if query.Remote {
+		if query.Format == define.OCIManifestDir {
```
Comment (rhatdan) [inline pkg/api/handlers/libpod/images.go:278]: Would it be better to just do this on the client side and not change the server side at all.

Comment (zhangguanzhang) [reply pkg/api/handlers/libpod/images.go:278]: Server will save to dir, not respone in on file.and client will untar the archive file

Comment (rhatdan) [reply pkg/api/handlers/libpod/images.go:278]: I am saying that the client checks to see if it is remote and sends across the 

		query.Format = define.OCIArchive
                 or
		query.Format = define.V2s2Archive

And then dumps it locally.

Comment (zhangguanzhang) [reply pkg/api/handlers/libpod/images.go:278]: > I am saying that the client checks to see if it is remote and sends across the
> 
> ```
> 	query.Format = define.OCIArchive
>              or
> 	query.Format = define.V2s2Archive
> ```
> 
> And then dumps it locally.
if format is an archive, will return at these lines
https://github.com/containers/podman/blob/df6c7c2ae5fb77a557d5616f8b422c45a43fe0ff/pkg/domain/infra/tunnel/images.go#L270-L280
if is a dir, will recreive an archive of the dir, and untar at this line
https://github.com/containers/podman/blob/df6c7c2ae5fb77a557d5616f8b422c45a43fe0ff/pkg/domain/infra/tunnel/images.go#L299
The logic already exists, so I don’t need to add it

File: pkg/bindings/images/images.go [hunk 3]
```diff
@@ -143,6 +143,7 @@ func Export(ctx context.Context, nameOrIDs []string, w io.Writer, options *Expor
 	for _, ref := range nameOrIDs {
 		params.Add("references", ref)
 	}
+	params.Add("remote", "true")
```
Comment (vrothberg) [inline pkg/bindings/images/images.go:146]: This parameter shouldn't be needed as it's always remote.

Comment (zhangguanzhang) [reply pkg/bindings/images/images.go:146]: Should I remove all the remote parameter?

Comment (zhangguanzhang) [reply pkg/bindings/images/images.go:146]: Should I remove all the remote parameter in all files?

Comment (vrothberg) [reply pkg/bindings/images/images.go:146]: Thanks, yes please remove it.

File: pkg/api/handlers/libpod/images.go [hunk 4]
```diff
@@ -270,6 +270,16 @@ func ExportImages(w http.ResponseWriter, r *http.Request) {
...
 
+	// if format is dir, server will save to an archive
+	// the client will unArchive after receive the archive file
+	// so must convert is at here
+	if query.Format == define.OCIManifestDir {
+		query.Format = define.OCIArchive
+	}
+	if query.Format == define.V2s2ManifestDir {
+		query.Format = define.V2s2Archive
+	}
```
Comment (vrothberg) [inline pkg/api/handlers/libpod/images.go:281]: Non-blocking: a `switch query.Format {...}` may be more idiomatic.

LGTM

Comment (zhangguanzhang) [reply pkg/api/handlers/libpod/images.go:281]: had done

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy test buttons

**Comment by rhatdan** [comment]
Someone will have to explain to me why this is being done on the server side?  Seems like the client knows what it is doing, and the server is not doing what the caller requested.  If I write a python client and ask for a directory, I will not have the majic available to convert, and will end up with a mess.  Or am I missing something?

**Comment by zhangguanzhang** [comment]
> Someone will have to explain to me why this is being done on the server side? Seems like the client knows what it is doing, and the server is not doing what the caller requested. If I write a python client and ask for a directory, I will not have the majic available to convert, and will end up with a mess. Or am I missing something?

server side always return file bytes in one file：
https://github.com/containers/podman/blob/6ff56ab50a10e2350013c4ff1bf4b8d5d7b5aa87/pkg/api/handlers/libpod/images.go#L230-L236

**Comment by rhatdan** [comment]
Well outvoted and it does fix a bug, so merging.
/lgtm
